truss/truss_config.py

```diff
@@ -106,6 +108,7 @@ def from_dict(d):
             bundled_packages_dir=d.get(
                 "bundled_packages_dir", DEFAULT_BUNDLED_PACKAGES_DIR
             ),
+            spec_version=d.get("spec_version", DEFAULT_SPEC_VERSION),
```

Comment:
There is a risk here, that we read an old truss, which will have spec_version missing, and we convert it to version 2, breaking the backward compatibility. I think we should set this here to `1.0`.  

There would still be an issue here, where if someone creates TrussConfig directly from an existing truss yaml, without using the from_dict or from_yaml method, they can still end up with 2, but that is very unlikely and I don't know how we can handle that case better.